projects/vdk-plugins/vdk-trino/src/vdk/plugin/trino/trino_plugin.py

```diff
@@ -88,14 +92,41 @@ def initialize_job(context: JobContext) -> None:
     )
 
 
-@hookimpl(hookwrapper=True, trylast=True)
+@hookimpl(hookwrapper=True, tryfirst=True)
 def run_step(context: JobContext, step: Step) -> None:
     out: pluggy.callers._Result
     out = yield
     if out.excinfo:
         exc_type, exc_value, exc_traceback = out.excinfo
         if isinstance(exc_value, TrinoUserError):
-            raise UserCodeError() from exc_value
+            raise UserCodeError(ErrorMessage()) from exc_value
+    if out.result:
+        step_result: StepResult = out.result
+        if isinstance(step_result.exception, requests.exceptions.ConnectionError):
+            raise VdkConfigurationError(
```

Comment:
Configuration Errors are treated as user errors when ran locally and as platform errors when run in cloud. 

There really should be called Environment error as they may not be just about configuration - e.g if there's a connectivity issue (be it because of misconfiguration or network outage) the platform is responsible to check it out when run on the "cloud" but the end-user would need to check it out when ran locally  (if the issue turns out to be in the server, they'd raise a ticket to the platform team).

Reply:
I see. Thanks